From 8c42a0c244cb06461f3b7405b73fdbc2b1b2616c Mon Sep 17 00:00:00 2001
From: Julian Brown <julian.brown@cpanel.net>
Date: Wed, 27 May 2020 12:36:51 -0500
Subject: [PATCH] These are only used with very old openssl

---
 src/osdep/unix/ssl_unix.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/osdep/unix/ssl_unix.c b/src/osdep/unix/ssl_unix.c
index a01769c..487258e 100644
--- a/src/osdep/unix/ssl_unix.c
+++ b/src/osdep/unix/ssl_unix.c
@@ -59,14 +59,15 @@ typedef struct ssl_stream {
 static SSLSTREAM *ssl_start(TCPSTREAM *tstream,char *host,unsigned long flags);
 static char *ssl_start_work (SSLSTREAM *stream,char *host,unsigned long flags);
 static int ssl_open_verify (int ok,X509_STORE_CTX *ctx);
+#if OPENSSL_VERSION_NUMBER < 0x10100000
 static char *ssl_validate_cert (X509 *cert,char *host);
 static long ssl_compare_hostnames (unsigned char *s,unsigned char *pat);
 static char *ssl_getline_work (SSLSTREAM *stream,unsigned long *size,
 			       long *contd);
+#endif
 static long ssl_abort (SSLSTREAM *stream);
 static RSA *ssl_genkey (SSL *con,int export,int keylength);
 
-
 /* Secure Sockets Layer network driver dispatch */
 
 static struct ssl_driver ssldriver = {
@@ -369,6 +370,7 @@ static char *ssl_validate_cert (X509 *cert,char *host)
  * Returns: T if pattern matches base, else NIL
  */
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000
 static long ssl_compare_hostnames (unsigned char *s,unsigned char *pat)
 {
   long ret = NIL;
@@ -460,6 +462,7 @@ static char *ssl_getline_work (SSLSTREAM *stream,unsigned long *size,
   else *contd = LONGT;		/* continuation needed */
   return ret;
 }
+#endif
 
 /* SSL receive buffer
  * Accepts: SSL stream
-- 
2.18.2

